File: 2026-09/src/day_12/prms_181859/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> stk;

    for (int x : arr) {
        if (stk.empty() || stk.back() != x) {
            stk.push_back(x);
        } else {
            stk.pop_back();
        }
    }

    if (stk.empty()) return {-1};
    return stk;
}

File: 2026-09/src/day_12/prms_181859/Solution.java
package day_12.prms_181859;

import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] stk = new int[arr.length];
        int idx = 0;

        for (int x : arr) {
            if (idx == 0) {
                stk[idx++] = x;
            } else {
                if (stk[idx - 1] == x) {
                    idx--;
                } else {
                    stk[idx++] = x;
                }
            }
        }

        if (idx == 0) return new int[]{-1};
        return Arrays.copyOf(stk, idx);
    }
}
